editor/selectors.js

```diff
@@ -748,6 +748,26 @@ export function isBlockSelected( state, uid ) {
 	return start === uid;
 }
 
+/**
+ * Returns true if the block corresponding to the specified unique ID is
+ * currently selected but isn't the last of the selected blocks. Here "last"
```

Comment:
This seems like a strangely unexpected behavior for a function called `isBlockWithinSelection`. On its name, I'd expect it to return `true` if the block is _anywhere_ in the current selection, regardless of multi-selection, singular selection, even considering block hierarchy (I arrive here from https://github.com/WordPress/gutenberg/pull/15499#issuecomment-491333277 and relevant [current logic of block list async rendering](https://github.com/WordPress/gutenberg/blob/94d48e09dcfb547ad1b142d137a51a378d57077f/packages/block-editor/src/components/block-list/index.js#L204-L223)). Excluding the last block in the multi-selection is odd, even if for the particular requirements of the UI it's a necessary exclusion.

Do you think it's something which would make sense to revise the behavior and account for the UI needs in some other way, like a separate selector call to test whether the selected clientId is last in a multi-selection?